File: server/models/userModel.js
const mongoose = require('mongoose')
const Schema = mongoose.Schema;
// Definice schématu pro model uživatele
const UserSchema = new mongoose.Schema(
    {
  name: {
    type: String,
    required: true
  },
  surname: {
    type: String,
    required: true
  },
  email: {
    type: String,
    unique: true,
    required: true
    
  },
  password: {
    type: String,
    required: true
  },
  bankAccounts: {
    _bankId: {
      type: Schema.Types.ObjectId
    },
    bankInfo: {
      type: String
    },
    bankNumber: {
      type: String
    }
  },
  brokerAccounts: {
    _brokerId: {
      type: Schema.Types.ObjectId
    },
    brokerInfo: {
      type: String
    },
    brokerLogin: {
      type: String
    },
    brokerPassword: {
      type: String
    }
  },
  incomes: {
    _incomeId: {
      type: Schema.Types.ObjectId
    },
    category: {
      type: String
    },
    amount: {
      type: Number
    },
    date: {
      type: Date
    },
    regular: {
      type: Boolean
    }
  },
  expenses: {
    _expenseId: {
      type: Schema.Types.ObjectId
    },
    category: {
      type: String
    },
    amount: {
      type: Number
    },
    date: {
      type: Date
    },
    regular: {
      type: Boolean
    }
  },
  investments: {
    _investmentId: {
      type: Schema.Types.ObjectId
    },
    ticker: {
      type: String
    },
    type: {
      type: String
    },
    amount: {
      type: Number
    },
    date: {
      type: Date
    },
    regular: {
      type: Boolean
    },
    isOpen: {
      type: Boolean
    },
    openingPrice: {
      type: Number
    },
    closingPrice: {
      type: Number
    }
  }
}, 
{ collection: 'users' } // Název kolekce (tabulky) v MongoDB, kam budou ukládána data
)
// Vytvoření modelu na základě definovaného schématu
const model = mongoose.model('UserSchema', UserSchema)

// Exportování vytvořeného modelu pro použití v jiných částech aplikace
module.exports = model

File: investNetCore/Models/UserModel.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System.ComponentModel.DataAnnotations;

namespace investNetCore.Models
{
    public class UserModel
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? Id { get; set; }

        [Required(ErrorMessage = "Prosím, vyplňte meno.")]
        public string name { get; set; } = null!;
        [Required(ErrorMessage = "Prosím, vyplňte Priezvisko.")]
        public string surname { get; set; } = null!;

        //[Required(ErrorMessage = "Prosím, vyplňte Váš email.")]
        public string email { get; set; } = null!;
        [Required(ErrorMessage = "Prosím, zvoľte si heslo.")]
        public string password { get; set; } = null!;
        public List<BankAccount> BankAccounts { get; set; }
        public List<BrokerAccount> BrokerAccounts { get; set; }
        public List<Income> Incomes { get; set; }
        public List<Expense> Expenses { get; set; }
        public List<Investment> Investments { get; set; }



    }
    public class BankAccount
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? _BankId { get; set; }
        public string BankInfo { get; set; } = null!;
        public decimal BankNumber { get; set; } 
    }
    public class BrokerAccount
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? _BrokerkId { get; set; }
        public string BrokerInfo { get; set; } = null!;
        public string BrokerLogin { get; set; } = null!;
        public string BrokerPassword { get; set; } = null!;


    }

    public class Expense
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? _ExpensekId { get; set; }

        public string Category { get; set; }
        public decimal Amount { get; set; }
        public DateTime Date { get; set; }
    }
    public class Income
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? _IncomekId { get; set; }

        public string Category { get; set; }
        public decimal Amount { get; set; }
        public DateTime Date { get; set; }
    }
    public class Investment
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? _InvestmentkId { get; set; }
        public string Ticker { get; set; }
        public Boolean Type { get; set; }
        public decimal Amount { get; set; }
        public DateTime Date { get; set; }
        public Boolean IsOpen { get; set; }
        public decimal OpeningPrice { get; set; }
        public decimal ClosingPrice { get; set; }
    }

}
